src/haz3lcore/zipper/action/Move.re

```diff
@@ -178,6 +178,31 @@ module Make = (M: Editor.Meta.S) => {
     jump_to_p(z, piece => Piece.id(piece) == id);
   };
 
+  let jump_to_indicated_var = (z: t): option(t) => {
+    open OptUtil.Syntax;
+    let* (piece, _, _) = Indicated.piece(z);
+    let* statics =
+      switch (piece) {
+      | Base.Tile(tile) =>
+        let (term, _) = MakeTerm.go(Zipper.unselect_and_zip(z));
+        Id.Map.find_opt(tile.id, Statics.mk_map(term));
+      | _ => None
+      };
+
+    let* info_exp =
+      switch (statics) {
+      | Statics.InfoExp(info_exp) => Some(info_exp)
+      | _ => None
+      };
+
+    let* entry =
+      switch (info_exp.term.term) {
+      | TermBase.UExp.Var(name) => VarMap.lookup(info_exp.ctx, name)
+      | _ => None
+      };
+    jump_to_id(z, entry.id);
+  };
+
```

Comment:
having thought about this a bit more, lets refactor this:
1. I want to avoid statics dependencies in Move. for now let's consolidate these in Perform.
2. Put the maketerm/mk_map calcs at the top of go_z in perform, so the map can be (hypothetically) re-used across cases, as well as the Indicated call. I forgot earlier but don't bother to call Indicated.piece, you can just do Indicated.index to get the id directly.
3. Make a function in Statics called something like get_binding_site which takes an id and a map and returns option Id; some if id points to an infor_exp which is a variable reference which exists in its context.
4. depending on how small this makes jump_to_indicated_var maybe just inline it; your discretion

Reply:
This all makes sense to me. Just want to call out that I'm not sure how expensive `maketerm` and `mk_map` are, so I'm not sure if this will slow down things like moving up/down.